.github/pull_request_template.md

```diff
@@ -4,3 +4,4 @@ Please make sure to fulfil the following conditions before marking this PR ready
 
 -   [ ] If this PR adds or changes features or fixes bugs, this has been added to the changelog
 -   [ ] If this PR adds new actions or other ways to alter the state, [test scenarios](https://github.com/hpi-sam/digital-fuesim-manv-public-test-scenarios) have been added
+-   [ ] I have the right to the code I commited in this Pull Request (this includes that I don't use any AI tools that violate or could violate the license [LICENSE-README.md](LICENSE-README.md) of this project or copied code from e.g. a forum that falls under an incompatible license and if compatible have given credit in the correct way).
```

Comment:
```suggestion
-   [ ] I have the right to license the code submitted with this Pull Request (i.e., this is my own code or code licensed under a license compatible to AGPL v3.0). In case of using third party code, I have given appropriate credit.
```

I think this a bit more concise without being as oddly specific (regarding AI tools or forums) as the current wording.

Reply:
I put AI tools into it, as many people probably don't know the problems that can occur when they for example use Githubs copilot (which may violate licenses). Of course our license is compatible with many licenses, but with such tools you probably won't know the source and therefore the attributions are missing.
Same with forums like stackoverflow, which also have corresponding licenses (which should be compatible mostly with ours, but still need attribution).

Also if I am thinking about it, I am not sure if we need to also include attributions in the html file (like scraping through every code file and looking where something like stackoverflow is mentioned). We probably don't "need" to, as e.g. GPL code doesn't need to attributed when it is sent over a network only, which is common for a website/webapp like ours. I am not a lawyer, so the whole third party acknowledgements could be just a nice gesture.
We may put some hint that some part of the code is inspired or copied from forums, such as stackoverflow on the top of the html or so?!

Reply:
> I put AI tools into it, as many people probably don't know the problems that can occur when they for example use Githubs copilot (which may violate licenses). [...] Same with forums like stackoverflow

I understand your motivation, but on the other hand, if we mention some sources explicitly, it could seem that other sources that are not mentioned are okay. And problems with AI tools, forums etc. are not specific to our project but apply to all sorts of (open source) software with any license, so I don't think this is something that we have to mention explicitly.

> Also if I am thinking about it, I am not sure if we need to also include attributions in the html file (like scraping through every code file and looking where something like stackoverflow is mentioned).

I think we should do this. Searching for URLs (e.g. with `https?://` regex) in our code files could be a good way to find such third party snippets might be an easy way to find them, as all the snippets I've encountered so far had their source in a comment above.

> We may put some hint that some part of the code is inspired or copied from forums, such as stackoverflow on the top of the html or so?!

I think for the typical extent of code from forums it would be fine to do it like this and keep the copyright notices in the comments of the code as a detailed reference.

Reply:
You are right (I would also add it applies probably even more to non open source code).
I can try to find a way to extract these and add them to the table.

Reply:
@lukasrad02 I looked at trying to use https:// finding, but it resulted in a lot of files (even when only searchin in the `src` folders). They would be many false positives, as there are many "this is a quick fix, see the following issue" or so.
Sometimes `inspired by` is used, but not always, sometimes `see` is used, but also not in a way that would be copyright related.
There are multiple ways to solve this.
1. We don't include them (which shouldn't be a problem legally, as long as it isn't a license similar to AGPL that also counts sending code over the network as distribution). But the same applies probably to all third-party notices right now. They are just like a nice gesture or some kind of thank you page.
2. Going through every `https://` and looking manually if they fall under copyright and rewrite it to something like `inspired by` or add some other wording that can be found.
3. Just include a list of all lines that include an `https://` in it

The second solution would need a bit of work and would mean that in the future everyone needs to follow it, so some guideline also, maybe.

I am using the following filtering right now in VSCode:
search: `https://`
replace: empty
files to include: `src/**`
files to exclude: `node_modules, dist, package-lock.json, package.json, THIRD-PARTY-LICENSE-ACKNOWLEDGEMENTS.html, THIRD-PARTY-LICENSE-ACKNOWLEDGEMENTS.json, *.md`

resulting in 112 results in 52 files.